sound/soc/sof/debug.c

```diff
@@ -626,6 +627,124 @@ int snd_sof_debugfs_buf_item(struct snd_sof_dev *sdev,
 }
 EXPORT_SYMBOL_GPL(snd_sof_debugfs_buf_item);
 
+static int memory_info_update(struct snd_sof_dev *sdev, char **out)
+{
+	struct sof_ipc_cmd_hdr msg = {
+		.size = sizeof(struct sof_ipc_cmd_hdr),
+		.cmd = SOF_IPC_GLB_DEBUG | SOF_IPC_DEBUG_MEM_USAGE,
+	};
+	struct sof_ipc_dbg_mem_usage *reply;
+	int len = 0;
+	char *buf;
+	int ret;
+	int i;
+
+	reply = kmalloc(SOF_IPC_MSG_MAX_SIZE, GFP_KERNEL);
+	if (!reply)
+		return -ENOMEM;
+
+	ret = pm_runtime_get_sync(sdev->dev);
```

Comment:
I wonder if the need to actually disable pm_runtime if you want to track memory usage.
On Intel SKL+ platforms, we power-gate the DSP after 5 seconds of idleness, so it'll be very have for an application to retrieve the information at specific points in time?

Reply:
> I wonder if the need to actually disable pm_runtime if you want to track memory usage.

Then it will be impossible to track memory leaks in low-power-state to run-state.

> On Intel SKL+ platforms, we power-gate the DSP after 5 seconds of idleness, so it'll be very have for an application to retrieve the information at specific points in time?

For memory leaks detection, application always needs to retrieve information at specific point in time. For memory occupation logging, it's not so critical.

Reply:
@ktrzcinx how would the application know about the 'specific point in time'? My point is that to detect memory leaks, you need to open, configure, stream, close. And then check if there's any memory footprint increase. However, 4s after the close the DSP is power gated so the information will be lost. If you resume, you will restart from zero with a firmware boot.

Reply:
> However, 4s after the close the DSP is power gated so the information will be lost.

It won't change after fully support for low power modes in SOF? Even if not, then this feature will be used mainly by test scripts, then 4s is enormous amount of time (enough to read memory info hundreds of times) and choosing right point in time shouldn't be any difficult (eg.  aplay start, aplay stop, read, aplay start, aplay stop, read).

Reply:
@ktrzcinx maybe with scripts, but it's unusable when checking interactively. I take more than 4 seconds to type a new command.... even for a script, we don't necessarily control timing precisely and this is error -prone.

Reply:
It sounds like a different feature, for separate PR. Preventing auto_suspend may have broader use-case than memory info scanning. Moreover, as I said it before, those two features shouldn't be strictly tighten, to allow testing low power modes more sophisticated than power down.

Reply:
Block PM functionality added in #2477